lib/internal/http2/core.js

```diff
@@ -1628,17 +1507,20 @@ function continueStreamDestroy(self, err, callback) {
     err = new errors.Error('ERR_HTTP2_STREAM_ERROR', code);
   }
   callback(err);
-  process.nextTick(emit, self, 'streamClosed', code);
-  debug(`[${sessionName(session[kType])}] stream ${self[kID]} destroyed`);
+  process.nextTick(emit, this, 'streamClosed', code);
+  debug(`[${sessionName(session[kType])}] stream ${this[kID]} destroyed`);
 }
 
-function finishStreamDestroy(self, handle) {
-  const id = self[kID];
-  self[kSession][kState].streams.delete(id);
-  delete self[kSession];
-  if (handle !== undefined)
-    handle.destroyStream(id);
-  self.emit('destroy');
+function finishStreamDestroy() {
+  const id = this[kID];
+  this[kSession][kState].streams.delete(id);
```

Comment:
Complete nit, but since `id` is now used only once, can we `delete(this[kID]);` instead?

Reply:
Maybe I’m missing context, but `delete` is always going to change the map of the object whose property is being delete, so you don’t usually want that for performance reasons if it’s not necessary.

(And if you delete anything but the most recently added property, you’re going to switch the object into dictionary mode, which is even worse.)

Reply:
Sorry, to be clear I just mean this: `this[kSession][kState].streams.delete(this[kId]);`. Avoid the unnecessary `const` declaration.

Reply:
@addaleax ... Keep in mind that this is Map, not a regular object.